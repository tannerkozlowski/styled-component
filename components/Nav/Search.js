import React from 'react'
import styled, { css, createGlobalStyle } from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import PropTypes from 'prop-types'
import {
  lightGrey,
  grey,
  lightVioletRed,
  gold,
  darkGrey,
} from '../../utils/colors'
import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import { resetInput } from '../../utils/form'
import { mobile } from '../../utils/media'

const INPUT_ID = 'docs-search-input'

const Wrapper = styled.form`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex: 0 0 auto;
  ${mobile(css`
    display: block;
    padding: ${rem(10)};
    span.algolia-autocomplete {
      display: block !important;
    }
    span.ds-dropdown-menu::before {
      display: none;
    }
  `)};
`

const Input = styled.input`
  ${resetInput};
  flex: 0 0 auto;
  width: ${rem(130)};
  line-height: ${rem(navbarHeight - 20)};
  font-size: ${rem(15)};
  color: currentColor;
  ::placeholder {
    color: currentColor;
    opacity: 0.5;
  }
  &:focus {
    border-bottom: 1px solid currentColor;
  }
  ${mobile(css`
    padding-left: ${rem(40)};
    display: block;
    width: 100%;
    border: 1px solid ${lightGrey};
    border-radius: ${rem(3)};
    background: ${grey};
    color: white;
    &:focus {
      border-bottom: 1px solid ${lightGrey};
    }
  `)};
`

const StyledSearchIcon = styled(FontAwesomeIcon).attrs({
  icon: faSearch,
})``

const Button = styled.label.attrs({
  for: INPUT_ID,
})`
  ${resetInput};
  flex: 0 0 auto;
  ${resetInput} flex: 0 0 auto;
  margin-right: ${rem(4)};
  cursor: pointer;
  display: flex;
  &:hover,
  &:focus {
    opacity: 0.7;
    filter: drop-shadow(0 1px 2px rgba(0, 0, 0, 0.2));
  }

  &:active {
    background: rgba(0, 0, 0, 0.1);
  }

  ${StyledSearchIcon} {
    width: ${rem(20)};
    height: ${rem(20)};

    path {
      fill: currentColor;
    }
  }
  ${mobile(css`
    position: absolute;
    height: ${rem(34)};
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${rem(34)};
    z-index: 1;
    background: ${grey};
    color: white;
    border-radius: ${rem(3)};
  `)};
`

const SearchButton = () => (
  <Button>
    <StyledSearchIcon />
  </Button>
)

const GlobalStyles = createGlobalStyle`
    .algolia-autocomplete
      .ds-dropdown-menu
      .ds-suggestion.ds-cursor
      .algolia-docsearch-suggestion.suggestion-layout-simple,
    .algolia-autocomplete
      .ds-dropdown-menu
      .ds-suggestion.ds-cursor
      .algolia-docsearch-suggestion:not(.suggestion-layout-simple)
      .algolia-docsearch-suggestion--content {
      background: ${lightVioletRed};
    }

    /* Main category (eg. Getting Started) */
    .algolia-autocomplete .algolia-docsearch-suggestion--category-header {
      color: ${darkGrey};
    }

    /* Category (eg. Downloads) */
    .algolia-autocomplete .algolia-docsearch-suggestion--subcategory-column {
      color: ${grey};
    }

    /* Title (eg. Bootstrap CDN) */
    .algolia-autocomplete .algolia-docsearch-suggestion--title {
      font-weight: bold;
      color: black;
    }

    /* Description description (eg. Bootstrap currently works...) */
    .algolia-autocomplete .algolia-docsearch-suggestion--text {
      color: ${grey};
    }

    /* Highlighted text */
    .algolia-autocomplete .algolia-docsearch-suggestion--highlight {
      color: ${gold};
      background: transparent;
    }
`

const Search = ({ isDocs, className }) => (
  <Wrapper className={className}>
    <GlobalStyles />
    <SearchButton />
    <Input
      id={INPUT_ID}
      placeholder={isDocs ? `Search ...` : `Search docs ...`}
    />
  </Wrapper>
)

Search.propTypes = {
  className: PropTypes.string,
}

export default Search
