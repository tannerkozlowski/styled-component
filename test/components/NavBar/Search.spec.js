import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Search from '../../../components/Nav/Search'

test('Search renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON()

  expect(tree).toMatchSnapshot()
})
