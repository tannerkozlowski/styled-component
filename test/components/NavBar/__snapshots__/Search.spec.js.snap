// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 7.222222222222222rem;
  line-height: 1.6666666666666667rem;
  font-size: 0.8333333333333334rem;
  color: currentColor;
}

.c3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c3::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c3::placeholder {
  color: currentColor;
  opacity: 0.5;
}

.c3:focus {
  border-bottom: 1px solid currentColor;
}

.c1 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 0.2222222222222222rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:hover,
.c1:focus {
  opacity: 0.7;
  -webkit-filter: drop-shadow(0 1px 2px rgba(0,0,0,0.2));
  filter: drop-shadow(0 1px 2px rgba(0,0,0,0.2));
}

.c1:active {
  background: rgba(0,0,0,0.1);
}

.c1 .c2 {
  width: 1.1111111111111112rem;
  height: 1.1111111111111112rem;
}

.c1 .c2 path {
  fill: currentColor;
}

@media (max-width:62.5em) {
  .c0 {
    display: block;
    padding: 0.5555555555555556rem;
  }

  .c0 span.algolia-autocomplete {
    display: block !important;
  }

  .c0 span.ds-dropdown-menu::before {
    display: none;
  }
}

@media (max-width:62.5em) {
  .c3 {
    padding-left: 2.2222222222222223rem;
    display: block;
    width: 100%;
    border: 1px solid rgba(20,20,20,0.1);
    border-radius: 0.16666666666666666rem;
    background: #282a36;
    color: white;
  }

  .c3:focus {
    border-bottom: 1px solid rgba(20,20,20,0.1);
  }
}

@media (max-width:62.5em) {
  .c1 {
    position: absolute;
    height: 1.8888888888888888rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 1.8888888888888888rem;
    z-index: 1;
    background: #282a36;
    color: white;
    border-radius: 0.16666666666666666rem;
  }
}

<form
  className="c0"
>
  <label
    className="c1"
    for="docs-search-input"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-search fa-w-16 c2 "
      data-icon="search"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </label>
  <input
    className="c3"
    id="docs-search-input"
    placeholder="Search docs ..."
  />
</form>
`;
